File: 800/1777A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int prev = sc.nextInt();
            
            int cnt = 0;
            
            n--;
            while(n-- > 0)
            {
                int curr = sc.nextInt();
                if((curr&1) == (prev&1))
                    cnt++;
                
                prev = curr;
            }
            
            System.out.println(cnt);
        }
    }
}

File: 800/1777A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        cin >> n;

        int prev;
        cin >> prev;

        int cnt = 0;
        n--;  // one element already read

        while (n-- > 0) 
        {
            int curr;
            cin >> curr;

            if ((curr & 1) == (prev & 1)) 
                cnt++;

            prev = curr;
        }

        cout << cnt << "\n";
    }

    return 0;
}
